src/StatusLine.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Interface representing a Value Object of the HTTP Status-Line, as
+ * specified in RFC 2616 and RFC 7231.
+ *
+ * PHP version 5.3
+ *
+ * @category StatusLine
+ *
+ * @package Teapot
+ *
+ * @author    Barney Hanlon <barney@shrikeh.net>
+ * @copyright 2013-2016 B Hanlon. All rights reserved.
+ * @license   MIT http://opensource.org/licenses/MIT
+ * @link      https://shrikeh.github.com/teapot
+ */
+namespace Teapot;
+
+use Teapot\StatusCodeException\InvalidStatusCodeException;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Interface representing a Value Object of the HTTP Status-Line, as
+ * specified in RFC 2616 and RFC 7231.
+ *
+ * PHP version 5.3
+ *
+ * @category StatusLine
+ *
+ * @package Teapot
+ *
+ * @author    Barney Hanlon <barney@shrikeh.net>
+ * @copyright 2013-2016 B Hanlon. All rights reserved.
+ * @license   MIT http://opensource.org/licenses/MIT
+ * @link      https://shrikeh.github.com/teapot
+ */
+class StatusLine
```

Comment:
Yeah I was running out of things to call them. I try and follow @ciaranmcnulty's advice and not call it that, but as we all know, "naming things is hard", and I had to be careful as I am trying to keep the language consistent with the underlying RFCs. As it is, I'm tempted to rename `reason` to `reasonPhrase` as that's what is in the RFCs. Coming up with other names for status code seems in this case unintuitive.

These interfaces are going to be used to aid in validation/parsing, in that I can find all the RFCs, reflect them to get the constants, and generate HTTP codes. Thought you might like to know why they are in there.